CI/Dockerfile

```
@@ -71,55 +73,39 @@ ENV CXX=clang++
 FROM compiler-${COMPILER} as external_deps
 
 # accessing gloabl ARGs in build stage
-ARG geant4_version
-ARG EMBREE_BRANCH
+ARG GEANT4
+ARG GEANT4_VERSION
+ARG EMBREE_VERSION
 ARG install_dir 
 ARG build_dir
-ARG ci_jobs
+ARG CI_JOBS
 ARG CXX
 ARG CC
 
-ENV geant4_basename=geant4-v${geant4_version}
+ENV geant4_basename=geant4-v${GEANT4_VERSION}
 ENV geant4_tarball=${geant4_basename}.tar.gz
-ENV geant4_shasum=2397eb859dc4de095ff66059d8bda9f060fdc42e10469dd7890946293eeb0e39
 ENV geant4_build_dir=${build_dir}/geant4
 ENV geant4_install_dir=${install_dir}/geant4
 
-RUN mkdir -p ${geant4_build_dir}/build && \
-    cd ${geant4_build_dir} && \
-    wget https://gitlab.cern.ch/geant4/geant4/-/archive/v${geant4_version}/${geant4_tarball} --no-check-certificate && \
-    tar -xzf ${geant4_tarball} && \
-    cd build && \
-    cmake ../${geant4_basename} -DCMAKE_INSTALL_RPATH=${geant4_install_dir}/lib \
-                                -DCMAKE_INSTALL_PREFIX=${geant4_install_dir} \
-                                -DGEANT4_USE_SYSTEM_EXPAT=OFF \
-                                -DCMAKE_BUILD_TYPE=Release \
-                                -DCMAKE_CXX_COMPILER=${CXX} \
-                                -DCMAKE_C_COMPILER=${CC} \
-                                -DBUILD_STATIC_LIBS=ON && \
-    make -j${ci_jobs} && \
-    make install && \
-    cd && \
-    rm -rf ${geant4_build_dir}
-
-
-ENV embree_install_dir=${install_dir}/embree
-ENV embree_build_dir=${build_dir}/embree
-
-# Clone and install Embree
-RUN mkdir -p ${embree_build_dir}/build && \
-    cd ${embree_build_dir} && \
-    git clone -b ${EMBREE_BRANCH} https://github.com/embree/embree && \
-    cd build && \
-    cmake ../embree -DCMAKE_INSTALL_PREFIX=${embree_install_dir} \
-                    -DEMBREE_TASKING_SYSTEM=INTERNAL \
-                    -DEMBREE_ISPC_SUPPORT=OFF \
-                    -DEMBREE_TUTORIALS=OFF \
-                    -DEMBREE_TBB_ROOT=/usr && \
-    make -j${ci_jobs} && \
-    make -j${ci_jobs} install && \
-    cd && \
-    rm -rf ${embree_build_dir}
+RUN if [ "${GEANT4}" = "ON" ]; then \
```

This can be changed to 
```suggestion
RUN if [ "${GEANT4_VERSION}" != "OFF" ]; then \
```
with above recommendations